File: packages/sdk/src/models/PrimerError.ts
export class PrimerError extends Error {
  errorId: string;
  errorCode?: string;
  description: string;
  recoverySuggestion?: string;
  diagnosticsId?: string;

  constructor(errorId: string, errorCode: string | undefined, description: string, recoverySuggestion: string | undefined, diagnosticsId: string | undefined) {
    super(description);
    this.errorId = errorId;
    this.errorCode = errorCode;
    this.description = description;
    this.recoverySuggestion = recoverySuggestion;
    this.diagnosticsId = diagnosticsId;
  }
}

File: android/src/main/java/com/primerioreactnative/extensions/PrimerError.kt
package com.primerioreactnative.extensions

import com.primerioreactnative.datamodels.PrimerErrorRN
import io.primer.android.domain.error.models.PrimerError

fun PrimerError.toPrimerErrorRN(): PrimerErrorRN =
    PrimerErrorRN(
        errorId = errorId,
        errorCode = errorCode,
        description = description,
        diagnosticsId = diagnosticsId,
        recoverySuggestion = recoverySuggestion,
    )
